### PR TITLE
Intelli-sense in extensions file

### PR DESCRIPTION
**Pull request by teintinu**
Intellisense for extensions.json providing installed extensions
from issue #12744

* Still suggesting examples. Useful when there are no installed extensions

### CODE REVIEW

File: extensions/configuration-editing/src/extension.ts [hunk 1]
```diff
@@ -61,11 +64,25 @@ function registerSettingsCompletions(): vscode.Disposable {
...
 
-function newSimpleCompletionItem(text: string, range: vscode.Range, description?: string): vscode.CompletionItem {
+function registerExtensionsCompletions(): vscode.Disposable {
+	return vscode.languages.registerCompletionItemProvider({ pattern: '**/extensions.json' }, {
+		provideCompletionItems(document, position, token) {
+			const location = getLocation(document.getText(), document.offsetAt(position));
+			const range = document.getWordRangeAtPosition(position) || new vscode.Range(position, position);
+			if (location.path[0] === 'recommendations') {
+				return vscode.extensions.all
+					.filter(e => !e.id.startsWith('vscode.'))
```
Comment (sandy081) [inline extensions/configuration-editing/src/extension.ts]: - You also need to exclude following extension which is packaged along with VS Code - `Microsoft.vscode-markdown`. This was getting shown in the intelli-sense.

- Intelli-sense should not show already entered extensions.

Comment (thr0w) [inline extensions/configuration-editing/src/extension.ts]: @sandy081 
* Excluded  `Microsoft.vscode-markdown`. Can I exclude all packages starting with `Microsoft.vscode-` ?
* Intelli-sense should not show already entered extensions: OK
* Maybe it's better remove: ExtensionsConfigurationSchema / defaultSnippets / Example ?

File: extensions/configuration-editing/src/extension.ts [hunk 2]
```diff
@@ -61,11 +64,32 @@ function registerSettingsCompletions(): vscode.Disposable {
...
 }
 
-function newSimpleCompletionItem(text: string, range: vscode.Range, description?: string): vscode.CompletionItem {
+function registerExtensionsCompletions(): vscode.Disposable {
+	return vscode.languages.registerCompletionItemProvider({ pattern: '**/extensions.json' }, {
+		provideCompletionItems(document, position, token) {
+			const location = getLocation(document.getText(), document.offsetAt(position));
+			const range = document.getWordRangeAtPosition(position) || new vscode.Range(position, position);
+			if (location.path[0] === 'recommendations') {
+				const config = vscode.workspace && vscode.workspace.getConfiguration('extensions.json');
```
Comment (sandy081) [inline extensions/configuration-editing/src/extension.ts]: I do not think`vscode.workspace.getConfiguration('extensions.json')` will get the extension recommendations. 
You might have to read the document and parse it. You already have document with text, so you can just use parse from `jsonc-parser` to get the existing recommendations.

Comment (thr0w) [inline extensions/configuration-editing/src/extension.ts]: Ok. It's my first try to do something in vscode source. Sorry. I pushed another try.

**Comment by sandy081** [comment]
LGTM. Thanks a lot for PR.

**Pull request merged**